packages/modal/src/ModalStyles.tsx

```diff
@@ -24,65 +24,78 @@ export const ModalBaseStyle = styled.div`
-  margin: 0;
   margin-bottom: ${({ $short, theme }) => ($short ? -theme.spaceMap.md : 0)}px;
-  padding: ${({ theme }) => theme.spaceMap.xl}px
-    ${({ theme }) => theme.spaceMap.xxl}px;
+  padding: ${({ theme }) => theme.spaceMap.xl}px;
   font-size: ${({ theme }) => theme.fontSizesMap.md}px;
   line-height: 1.5em;
 
   ${({ theme }) => theme.mediaQueries.md} {
-    padding: ${({ theme }) => theme.spaceMap.xl}px
-      ${({ theme }) => theme.spaceMap.lg}px;
+    padding: ${({ theme }) => theme.spaceMap.lg}px;
   }
 `
 
 export const ModalTitleStyle = styled.div<{
   $center: boolean
   $withTitleIcon: boolean
+  $withCloseButton: boolean
+  $withBackButton: boolean
 }>`
-  font-size: ${({ theme }) => theme.fontSizesMap.md}px;
+  font-size: ${({ theme }) => theme.fontSizesMap.sm}px;
   line-height: 1.5em;
-  font-weight: 800;
-  margin-right: auto;
-  padding-right: 32px;
-  padding-left: ${({ $center }) => ($center ? '32px' : '0px')};
-  padding-top: ${({ $withTitleIcon }) => ($withTitleIcon ? '40px' : '0')};
-  margin-bottom: -20px;
+  font-weight: 700;
+  margin-left: ${({ theme, $center, $withBackButton }) =>
+    $center && !$withBackButton ? theme.spaceMap.xxl : '0'}px;
+  margin-right: ${({ theme, $center, $withCloseButton }) =>
+    $center && !$withCloseButton ? theme.spaceMap.xxl : '0'}px;
+  padding-top: ${({ theme, $withTitleIcon }) =>
+    $withTitleIcon ? theme.spaceMap.sm : '0'}px;
+  padding-left: ${({ theme }) => theme.spaceMap.sm}px;
+  padding-right: ${({ theme }) => theme.spaceMap.sm}px;
   flex-grow: 1;
+  align-self: center;
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    padding-left: ${({ theme }) => theme.spaceMap.xs}px;
+    padding-right: ${({ theme }) => theme.spaceMap.xs}px;
+  }
+`
+
+export const ModalTitleIconStyle = styled.div<{
+  $center: boolean
+}>`
+  line-height: 0.7;
+  margin-bottom: ${({ theme }) => theme.spaceMap.md}px;
 
-  ${({ $center, $withTitleIcon }) =>
+  ${({ $center }) =>
     $center &&
-    $withTitleIcon &&
     `
     display: flex;
-    flex-direction: column;
-    align-items: center;
+    justify-content: center;
   `}
 `
 
-export const ModalTitleIconStyle = styled.div`
-  padding-bottom: ${({ theme }) => theme.spaceMap.xl}px;
+export const ModalTitleTextStyle = styled.div`
+  margin: ${({ theme }) => theme.spaceMap.xs}px 0;
 `
 
-export const ModalCloseStyle = styled(ButtonIcon).attrs({
-  icon: <Close />,
-  color: 'secondary',
-  variant: 'ghost',
-  size: 'xs',
-})`
-  flex-shrink: 0;
-  margin: 0 -10px 0 -22px;
-  border-radius: 50%;
+export const ModalSubtitleStyle = styled.div`
+  color: ${({ theme }) => theme.colors.textSecondary};
+  font-size: ${({ theme }) => theme.fontSizesMap.xs}px;
+  font-weight: 400;
+  line-height: 24px;
+  margin-top: -${({ theme }) => theme.spaceMap.xl}px;
+  padding-top: 0;
+  padding-bottom: ${({ theme }) => theme.spaceMap.sm}px;
+  padding-left: ${({ theme }) => theme.spaceMap.xxl}px;
+  padding-right: ${({ theme }) => theme.spaceMap.xxl}px;
```

Comment:
**Not critical, only as a suggestion.** Maybe it would be more clear for readability
```suggestion
export const ModalSubtitleStyle = styled.div(({ theme }) => {
  const { fontSizesMap, spaceMap, colors } = theme;

  return {
    color: colors.textSecondary,
    fontSize: `${fontSizesMap.xs}px`,
    fontWeight: 400,
    lineHeight: '24px',
    marginTop: `-${spaceMap.xl}px`,
    padding: `0 ${spaceMap.xxl}px ${spaceMap.sm}px`,
  };
```

But don't wont change `styled-component`s code-style to `jss`. That's why another alternative:

```suggestion
export const ModalSubtitleStyle2 = styled.div`
  ${({ theme: { spaceMap, fontSizesMap, colors } }) => css`
    color: ${colors.textSecondary};
    font-size: ${fontSizesMap.xs}px;
    font-weight: 400;
    line-height: 24px;
    margin-top: -${spaceMap.xl}px;
    padding: 0 ${spaceMap.xxl}px ${spaceMap.sm}px;
  `}
`;
```

**P.S. We can write it as one of improvements for ui lib, and refactor in future.**